File: backend/src/main/java/outcobra/server/model/Timetable.java
package outcobra.server.model;

import java.util.List;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Timetable {
    @Id
    private Long id;

    @NotNull
    @OneToOne(mappedBy = "timetable")
    private Semester semester;

    @OneToMany(mappedBy = "timetable")
    private List<TimetableEntry> entries;
}

File: backend/src/main/kotlin/outcobra/server/model/domain/Timetable.kt
package outcobra.server.model.domain

import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.OneToMany
import javax.persistence.OneToOne
import javax.validation.constraints.NotNull

@Entity
data class Timetable(@NotNull @OneToOne(mappedBy = "timetable") var semester: Semester? = null,
                     @OneToMany(mappedBy = "timetable") var entries: List<TimetableEntry> = listOf())
    : ParentLinked, AbstractEntity() {

    override val parent: ParentLinked?
        get() = semester
}